cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -1668,6 +1668,82 @@ struct ListImpl : public Selection<ListImpl<Type>, Type> {
   }
 };
 
+struct DenseUnionImpl : public Selection<DenseUnionImpl, DenseUnionType> {
+  using Base = Selection<DenseUnionImpl, DenseUnionType>;
+  LIFT_BASE_MEMBERS();
+
+  Int32Builder value_offset_builder;
+  Int8Builder child_id_builder;
```

Comment:
It's worth noting that ArrayBuilder always produces a null bitmap, which is not necessary for the buffers these are used to produce. See also: TypedBufferBuilder